training/src/main/scala/com/airbnb/aerosolve/training/pipeline/NDTreePipeline.scala

```diff
@@ -141,7 +148,19 @@ object NDTreePipeline {
         val result: ((String, String), Either[NDTreeModel, FeatureStats]) = a match {
           case Left(y) => {
             // build tree
-            val model = NDTree(paramsBC.value.options, y.toArray).model
+            val dimension = y(0).length
```

Comment:
there is no significance of the first element, right? You are just peeking in a random node to find out the dimension?